File: Ej 1/src/isp1/FactoriaMontana.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package isp1;

import java.util.Random;

/**
 *
 * @author joy11
 */
public class FactoriaMontana implements FactoriaCarreraYBicicleta{
    
    @Override
    public Carrera crearCarrera(){
        return new CarreraMontana();
    };
    
    @Override
    public Bicicleta crearBicicleta(){
        BicicletaMontana bici = new BicicletaMontana();
        Random r = new Random();
        bici.ID = r.nextInt();
        
        return bici;
    }
    
}

File: Ej 2/factoria_montana.py
from factoria_carrera_bicicleta import FactoriaCarreraYBicicleta
from carrera_montana import CarreraMontana
from bicicleta_montana import BicicletaMontana
import random

class FactoriaMontana(FactoriaCarreraYBicicleta):
    
    def crearCarrera(self):
        return CarreraMontana()
    
    def crearBicicleta(self):
        bici = BicicletaMontana(ID=random.randint(0, 10000))
        return bici
